### ISSUE TITLE
Question: mAP for weights in backup dir

### ISSUE THREAD

**Issue by skaldesh** [AlexeyAB/darknet#3550]
Hey everyone.  
During the training, many snapshots are added to the backup dir. Once the training is done, the README suggests to run each weight against a validation set and compare the mAP (and maybe the IoU) of each weight to find the best.  

My question: Is this the only option, or is the performance of each weight recorded during the training (if I supply my validation set also during the training)? Like, is there some output on the command line for it?  
The README states you can print out at the end a graph that shows this for all weights. But a picture does not help me (I want to automate this process)

**Comment by AlexeyAB** [comment]
@skaldesh Hi,

There are only ways:
* use `_best.weights` if you didn't interrupt training
* look at generated `chart.png` with Loss & mAP
* check the mAP for each file (may be in loop by using bash/cmd script)

**Comment by skaldesh** [comment]
Hi @AlexeyAB , thanks for the quick response! 

> use _best.weights if you didn't interrupt training  
Wait, so you tell me that if I let the training run until the end and it terminates, I will have a file `_best.weights` that will be the best one? Even if later models are overfitting?

**Comment by keko950** [comment]
You can add this code https://github.com/AlexeyAB/darknet/pull/3465 to your darknet source code in order to save always your best mAP score between different sessions! I'm using it and works fine

**Comment by skaldesh** [comment]
@keko950 Thank you for that, looks like what I need. Just to be very clear, with your PR, darknet saves the best.weights it found so far, so that after interrupting the training, it knows in the next training session which were the best weights, before being interrupted?

**Comment by keko950** [comment]
> @keko950 Thank you for that, looks like what I need. Just to be very clear, with your PR, darknet saves the best.weights it found so far, so that after interrupting the training, it knows in the next training session which were the best weights, before being interrupted?

Yes, just before start training if a best weights its found it will map first it and save it on memory to compare with the next trainings!

**Comment by skaldesh** [comment]
Great, hope your PR gets merged soon
